### PR TITLE
Add new parameter --tumor-purity to control clonal vaf cutoff during aggregate report creation

### PR DESCRIPTION
**Pull request by susannasiebert**
This is to address https://github.com/griffithlab/pVACtools/issues/606#issuecomment-1030126576

### CODE REVIEW

File: pvactools/lib/aggregate_all_epitopes.py [hunk 1]
```diff
@@ -211,9 +217,12 @@ def get_list_unique_mutation_keys(self):
         return sorted(keys)
 
     def calculate_clonal_vaf(self):
-        vafs = np.sort(pd.read_csv(self.input_file, delimiter="\t", usecols=["Tumor DNA VAF"])['Tumor DNA VAF'].unique())[::-1]
-        vaf_clonal = list(filter(lambda vaf: vaf < 0.6, vafs))[0]
-        return vaf_clonal
+        if self.tumor_purity:
+            return self.tumor_purity * 0.5
+        else:
+            vafs = np.sort(pd.read_csv(self.input_file, delimiter="\t", usecols=["Tumor DNA VAF"])['Tumor DNA VAF'].unique())[::-1]
```
Comment (chrisamiller) [inline pvactools/lib/aggregate_all_epitopes.py:223]: A comment here might be nice to explain it.  
#if no tumor purity is provided, make a rough estimate by taking the list of VAFs < 0.6 (assumption is that these are CN-neutral) and return the largest as the marker of the founding clone

File: pvactools/lib/run_argument_parser.py [hunk 2]
```diff
@@ -245,6 +245,12 @@ def __init__(self):
             default=False,
             action='store_true'
         )
+        self.parser.add_argument(
+            "--tumor-purity",
+            help="Value between 0 and 1 indicating the fraction of tumor cells in the tumor sample. Information is used during aggregate report creation for a simple estimation whether variants are subclonal or clonal based on VAF.",
```
Comment (chrisamiller) [inline pvactools/lib/run_argument_parser.py:250]: Suggested change (here and below): 
help="Value between 0 and 1 indicating the fraction of tumor cells in the tumor sample. Information is used during aggregate report creation for a simple estimation of whether variants are subclonal or clonal based on VAF. If not provided, purity is estimated directly from the VAFs"

File: pvactools/tools/pvacseq/generate_aggregated_report.py [hunk 3]
```diff
@@ -19,17 +19,28 @@ def define_parser():
         "output_file",
         help="The file path to write the aggregated report tsv to"
     )
+    parser.add_argument(
+        "--tumor-purity",
+        help="Value between 0 and 1 indicating the fraction of tumor cells in the tumor sample. Information is used for a simple estimation whether variants are subclonal or clonal based on VAF.",
```
Comment (chrisamiller) [inline pvactools/tools/pvacseq/generate_aggregated_report.py:24]: Suggested change:
help="Value between 0 and 1 indicating the fraction of tumor cells in the tumor sample. Information is used during aggregate report creation for a simple estimation of whether variants are subclonal or clonal based on VAF. If not provided, purity is estimated directly from the VAFs"

File: pvactools/lib/run_argument_parser.py (hunk 2 above)
Comment (susannasiebert) [inline pvactools/lib/run_argument_parser.py:250]:
```suggestion
            help="Value between 0 and 1 indicating the fraction of tumor cells in the tumor sample. Information is used during aggregate report creation for a simple estimation of whether variants are subclonal or clonal based on VAF. If not provided, purity is estimated directly from the VAFs.",
```

File: pvactools/tools/pvacseq/generate_aggregated_report.py (hunk 3 above)
Comment (susannasiebert) [inline pvactools/tools/pvacseq/generate_aggregated_report.py:24]:
```suggestion
        help="Value between 0 and 1 indicating the fraction of tumor cells in the tumor sample. Information is used during aggregate report creation for a simple estimation of whether variants are subclonal or clonal based on VAF. If not provided, purity is estimated directly from the VAFs.",
```

File: pvactools/lib/aggregate_all_epitopes.py [hunk 4]
```diff
@@ -211,9 +217,12 @@ def get_list_unique_mutation_keys(self):
         return sorted(keys)
 
     def calculate_clonal_vaf(self):
-        vafs = np.sort(pd.read_csv(self.input_file, delimiter="\t", usecols=["Tumor DNA VAF"])['Tumor DNA VAF'].unique())[::-1]
-        vaf_clonal = list(filter(lambda vaf: vaf < 0.6, vafs))[0]
-        return vaf_clonal
+        if self.tumor_purity:
+            return self.tumor_purity * 0.5
+        else:
```
Comment (susannasiebert) [inline pvactools/lib/aggregate_all_epitopes.py:222]:
```suggestion
        else:
        #if no tumor purity is provided, make a rough estimate by taking the list of VAFs < 0.6 (assumption is that these are CN-neutral) and return the largest as the marker of the founding clone
```

**Review by chrisamiller — APPROVED** [review]
+1

**Pull request merged**